src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -141,7 +141,7 @@ private bool TryReadToSlow(out ReadOnlySequence<T> sequence, T delimiter, T deli
                 else
                 {
                     // No delimiter, need to check the end of the span for odd number of escapes then advance
-                    if (remaining[remaining.Length - 1].Equals(delimiterEscape))
```

Comment:
Can this be !End?

Reply:
Also, can you add a test for this?

Reply:
Sure, will do, but I think this API still needs more work as part of  https://github.com/dotnet/corefxlab/issues/2513

For example, there is potentially unnecessary work happening if `remaining.Length == 1`.

Reply:
Yeah, I just don't want to ever fix a bug we find without making sure we add a test at this point.

Reply:
> Yeah, I just don't want to ever fix a bug we find without making sure we add a test at this point.

Agreed :)